src/main/java/com/backoffice/upjuyanolja/domain/coupon/entity/DiscountType.java

```diff
@@ -0,0 +1,112 @@
+package com.backoffice.upjuyanolja.domain.coupon.entity;
+
+import static com.backoffice.upjuyanolja.domain.coupon.entity.DiscountRestrictions.*;
+import com.backoffice.upjuyanolja.domain.coupon.exception.WrongCouponInfoException;
+import java.text.DecimalFormat;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import lombok.Getter;
+
+@Getter
+public enum DiscountType {
+
+    FLAT(
+        "원 할인 쿠폰",
+        "원 쿠폰",
+        (titleName, discount) -> {
+            DecimalFormat decimalFormat = new DecimalFormat("#,###");
+            String formattedNumber = decimalFormat.format(discount);
+            StringBuilder sb = new StringBuilder(formattedNumber);
+            sb.append(titleName);
+            return sb.toString();
+        },
+        (listName, discount) -> {
+            StringBuilder sb = new StringBuilder(discount);
+            sb.append(listName);
+            return sb.toString();
+        },
+        discount -> {
+            if (discount.intValue() >= getMinPrice() && discount.intValue() <= getMaxPrice()
+                    && discount.intValue() % 1000 == 0) {
+                return true;
+            }
+            throw new WrongCouponInfoException();
+        },
+        (price, discount) -> price.intValue() - discount.intValue()
+    ),
+
+    RATE(
+        "%할인 쿠폰",
+        "%쿠폰",
```

Comment:
별건 아니지만 % 뒤에 한 칸 공백이 있으면 좀 더 보기 편할 것 같습니다 ㅎㅎ
```suggestion
        "% 할인 쿠폰",
        "% 쿠폰",
```

Reply:
현재 디자인에서는 공백 없이 붙여서 출력하도록 되어 있습니다. 변경하려면 기획, FE와 협의가 필요한 부분입니다.

Reply:
디자인 최종안에 다음과 같이 공백이 있습니다! ㅎㅎ

![image](https://github.com/Upjuyanolja/Upjuyanolja_BE/assets/94631526/2afe33e7-c233-4036-bf09-6873b149323b)
![image](https://github.com/Upjuyanolja/Upjuyanolja_BE/assets/94631526/0d92713a-8938-4a81-bb69-dbe1a5121bf3)
![image](https://github.com/Upjuyanolja/Upjuyanolja_BE/assets/94631526/a0ddc5de-e862-468d-ad10-c70833b0a28c)

Reply:
디자인이 그새 또 바뀌었군요. 띄어쓰기 수정되었습니다.